src/full/Agda/Interaction/Imports.hs

```diff
@@ -379,6 +371,25 @@ addImportedThings isig metas ibuiltin patsyns display userwarn
   stOpaqueBlocks         `modifyTCLens` \ imp -> imp `Map.union` oblock
   stOpaqueIds            `modifyTCLens` \ imp -> imp `Map.union` oid
 
+-- | Merges two signatures, assuming the second is being imported into the first
+--
+-- This is not commutative, because we assume rewrite rules in the second
+-- signature cannot refer to definitions in the first.
```

Comment:
Maybe the comment could say something about the bias.  `Map.union` is left-biased so `importSignature` inherits this bias.
So, yes, it is not commutative.

I am surprised about the "cannot":  If I import both A and B and B also imports A, can't rewrite rules in B refer to symbols in A (which are then already in my imported signature)?